### PR TITLE
hw-jv-stream-github-practice

### PR DESCRIPTION
**Pull request by ihor-sydorenko**

### CODE REVIEW

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -14,7 +19,14 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(string -> string.split(","))
+                .flatMap(Arrays::stream)
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:24]:
```suggestion
                .flatMap(n -> Arrays.stream(n.split(",")))
```
image

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -35,7 +51,11 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
      */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getSex().equals(Person.Sex.MAN)
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:55]: image

File: src/main/java/practice/CandidateValidator.java [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE_FOR_CANDIDATE = 35;
+    private static final String NATIONALITY_FOR_CANDIDATE = "Ukrainian";
+    private static final int MIN_PERIOD_IN_UKRAINE = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] splitDatePeriod = candidate.getPeriodsInUkr().split("-");
+        int periodInUkraine = Integer.parseInt(splitDatePeriod[1])
+                - Integer.parseInt(splitDatePeriod[0]);
```
Comment (aaananas) [inline src/main/java/practice/CandidateValidator.java:15]: add constants for indexes

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -59,7 +85,14 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> !person.getCats().isEmpty()
+                        && person.getSex().equals(Person.Sex.WOMAN)
+                        && person.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```
Comment (aaananas) [inline src/main/java/practice/StreamPractice.java:93]:
```suggestion
                .flatMap(person -> person.getCats().stream())
```

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/practice/StreamPractice.java [hunk 5]
```diff
@@ -50,7 +68,14 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getSex().equals(Person.Sex.MAN)
+                        && person.getAge() >= fromAge
+                        && person.getAge() <= maleToAge
+                        || person.getSex().equals(Person.Sex.WOMAN)
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:75]: You still have this mistake(((
image

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and **correct all the points** described there. The mentor will not check the work until the checklist points are corrected

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**